File: code/python-for-csharp-java/compare/l03_numbers.cs
// compare/l03_numbers.cs
// Integer division truncates toward zero, and % takes the sign of the dividend
Console.WriteLine($"{7 / 2} {-7 / 2} {-7 % 2}");

// Math.Round rounds half to even by default too; decimal is exact for decimal fractions
Console.WriteLine($"{Math.Round(2.5)} {Math.Round(3.5)} {Math.Round(2.5, MidpointRounding.AwayFromZero)}");
Console.WriteLine($"{0.1 + 0.2} {0.1m + 0.2m}");

// int has 32 bits: without checked, it wraps around
int max = int.MaxValue;
Console.WriteLine(max + 1);

File: code/python-for-csharp-java/compare/L03Numbers.java
// compare/L03Numbers.java
class L03Numbers {
    public static void main(String[] args) {
        // / and % truncate toward zero; Math.floorDiv and Math.floorMod round like Python's // and %
        System.out.println((7 / 2) + " " + (-7 / 2) + " " + (-7 % 2) + " " + Math.floorDiv(-7, 2) + " " + Math.floorMod(-7, 2));

        // Math.round rounds half up; Math.rint rounds half to even
        System.out.println(Math.round(2.5) + " " + Math.round(3.5) + " " + Math.rint(2.5));
        System.out.println(0.1 + 0.2);

        // int has 32 bits and wraps around
        int max = Integer.MAX_VALUE;
        System.out.println(max + 1);
    }
}
